### PR TITLE
Adds protection against name collision to SoA internal variables.

### PR DESCRIPTION
**Pull request by ericcano**
#### PR description:

This PR simply add `_soa_impl_`  to internal implementation variables of SoA templates to limit the risk of collisions with user names/

#### PR validation:

The SoA tests (and Pixel code which also depends one SoA) compiles and the SoA unit tests run. The main risk was a failure to compile.

### CODE REVIEW

File: DataFormats/SoATemplate/interface/SoAView.h [hunk 1]
```diff
@@ -554,7 +554,7 @@ namespace cms::soa {
       Metadata(const Metadata&) = delete;                                                                              \
                                                                                                                        \
     private:                                                                                                           \
-      SOA_HOST_DEVICE SOA_INLINE Metadata(const VIEW& parent) : parent_(parent) {}                                     \
+      SOA_HOST_DEVICE SOA_INLINE Metadata(const VIEW& _soa_impl_parent) : parent_(_soa_impl_parent) {}                 \
```
Comment (fwyzard) [inline DataFormats/SoATemplate/interface/SoAView.h:557]: are you sure we need to protect _all_ these names ?
for example, I don't think this one can clash ?

Comment (ericcano) [reply DataFormats/SoATemplate/interface/SoAView.h:557]: Indeed, this one was over the top (removed).

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
+heterogeneous

**Comment by rappoccio** [comment]
+1

**Pull request merged**